### PR TITLE
bug: add ignored test for unexpected non-null when behavior

### PR DESCRIPTION
**Pull request by jordanrfrazier**
Adds an ignored test for unexpected behavior in the `when` function.

https://github.com/kaskada-ai/kaskada/issues/292

### CODE REVIEW

File: crates/sparrow-main/tests/e2e/when_tests.rs [hunk 1]
```diff
@@ -34,6 +34,57 @@ async fn when_data_fixture() -> DataFixture {
...
+    2023-01-01,Davor
+    2023-02-07,Ben
+   "},
+        )
+        .await
+        .unwrap()
+}
+
+#[tokio::test]
+#[ignore = "unexpected non-null behavior from when: https://github.com/kaskada-ai/kaskada/issues/292"]
```
Comment (bjchambers) [inline crates/sparrow-main/tests/e2e/when_tests.rs:78]: Suggest the ignore text being just the issue (easier to search for, etc.)

**Review by bjchambers — APPROVED** [review]
